bayesian/auth.py

```diff
@@ -26,3 +29,40 @@ def get_access_token(service_name):
 
     except Exception:
         logger.error('Unable to connect to Auth service')
+
+
+def get_user_type(view):
+    """Validate and get user type based on UUID from the request."""
+    @wraps(view)
+    def wrapper(*args, **kwargs):
+        """Read uuid and decides user type based on its validity."""
+        # Rule of UUID validation and setting user status ::
+        #  ==============================================================
+        #   UUID in request | UUID in RDS | RDS User State | User Status
+        #  ==============================================================
+        #    MISSING        | -- NA --    | -- NA --       | FREE
+        #    PRESENT        | MISSING     | -- NA --       | FREE
+        #    PRESENT        | PRESENT     | REGISTERED     | REGISTERED
+        #    PRESENT        | PRESENT     | !REGISTERED    | FREE
+        #  ==============================================================
+
+        # By default set this to 'freetier'.
+        g.registration_status = RegistrationStatus.freetier
+
+        # Read uuid from request header.
+        uuid = request.headers.get('uuid', None)
+
+        # Check user with uuid if it is present in RDS.
+        if uuid is not None:
+            try:
+                # Read user details from RDS based on uuid
+                user = get_user(uuid)
+                if user.status == 'REGISTERED':
+                    g.registration_status = RegistrationStatus.registered
+            except Exception as e:
+                logger.warning('User get for UUID: %s failed with %s', uuid, e)
+
+        logger.debug('For UUID: %s, got user type: %s', uuid, g.registration_status)
```

Comment:
Try block should have a block which would really generate an Exception.

```suggestion
if uuid:
    try:
        user = get_user(uuid)
    except <ProperExceptionClass_NOT_GenericException> e:
        logger.warning("Unable to get status for uuid=%s, err=%s", uuid, e)
    else:
        g.registration_status = RegistrationStatus[user.status]
     logger.debug('For UUID: %s, got user type: %s', uuid, g.registration_status)
```